support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/web/flow/DestroyTgtAndCookiesAction.java

```diff
@@ -0,0 +1,85 @@
+package org.apereo.cas.support.pac4j.web.flow;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apereo.cas.CentralAuthenticationService;
+import org.apereo.cas.configuration.model.core.logout.LogoutProperties;
+import org.apereo.cas.logout.LogoutRequest;
+import org.apereo.cas.web.flow.TerminateSessionAction;
+import org.apereo.cas.web.support.CookieRetrievingCookieGenerator;
+import org.apereo.cas.web.support.WebUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.webflow.action.EventFactorySupport;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+import com.google.common.base.Throwables;
+
+
+/**
+ * A light version of {@link TerminateSessionAction} that does NOT destroy the HTTP session, only destroys the TGT and cookies.
```

Comment:
Delete this class altogether. Simply introduce a new property and use that in the original action to do what you need. Set the setting to true for your deployment.

Reply:
Yes, you are right! We originally created this class for our customization because we could not touch TerminateSessionAction. Now we can.
I added a boolean property that can be configured via a setter. It will be set from the PAC4J configurator.
If you meant adding it to LogoutProperties, that would also be possible, I'm just not sure if I can then modify it at runtime from the PAC4J configurator. Because I wouldn't like the deployer to have to do that in properties, this can be done automatically.
The corresponding commit has been pushed.